modules/apps/packet-forward-middleware/ibc_middleware.go

```diff
@@ -238,26 +242,16 @@ func (im IBCMiddleware) OnRecvPacket(ctx sdk.Context, channelVersion string, pac
 // receiveFunds receives funds from the packet into the override receiver
 // address and returns an error if the funds cannot be received.
 func (im IBCMiddleware) receiveFunds(ctx sdk.Context, channelVersion string, packet channeltypes.Packet, data transfertypes.FungibleTokenPacketData, overrideReceiver string, relayer sdk.AccAddress) error {
-	overrideData := transfertypes.FungibleTokenPacketData{
-		Denom:    data.Denom,
-		Amount:   data.Amount,
-		Sender:   data.Sender,
-		Receiver: overrideReceiver, // override receiver
-		// Memo explicitly zeroed
-	}
-	overrideDataBz := transfertypes.ModuleCdc.MustMarshalJSON(&overrideData)
-	overridePacket := channeltypes.Packet{
-		Sequence:           packet.Sequence,
-		SourcePort:         packet.SourcePort,
-		SourceChannel:      packet.SourceChannel,
-		DestinationPort:    packet.DestinationPort,
-		DestinationChannel: packet.DestinationChannel,
-		Data:               overrideDataBz, // override data
-		TimeoutHeight:      packet.TimeoutHeight,
-		TimeoutTimestamp:   packet.TimeoutTimestamp,
+	data.Receiver = overrideReceiver
+	data.Memo = "" // Memo explicitly emptied.
+
+	overrideDataBz, err := json.Marshal(data)
+	if err != nil {
+		return err
 	}
 
-	ack := im.app.OnRecvPacket(ctx, channelVersion, overridePacket, relayer)
+	packet.Data = overrideDataBz // Override data.
```

Comment:
I'd rather not modify the argument value itself.
Instead, just create a copy (since this is not a pointer, this should work):
```go
overridePacket := packet
```

It also makes it clearer that you are in fact sending in an modified packet and not the original packet